### PR TITLE
 Speed up CQuorumManager::ScanQuorums

### PR DESCRIPTION
**Pull request by codablock**
See individual commits.

### CODE REVIEW

File: src/llmq/quorums.cpp [hunk 1]
```diff
@@ -308,26 +308,44 @@ std::vector<CQuorumCPtr> CQuorumManager::ScanQuorums(Consensus::LLMQType llmqTyp
...
+    auto firstQuorumHash = quorumBlockProcessor->GetFirstMinedQuorumHash(llmqType);
+    if (firstQuorumHash.IsNull()) {
+        // no quorum mined yet, avoid scanning the whole chain down to genesis
+        return {};
     }
 
-    result.reserve(maxCount);
+    CBlockIndex* pindex;
+    {
+        LOCK(cs_main);
```
Comment (UdjinM6) [inline src/llmq/quorums.cpp]: need `cs_main` for `GetQuorum()`

Comment (codablock) [reply src/llmq/quorums.cpp]: Whoops :see_no_evil: Fixed.

File: src/llmq/quorums.cpp [hunk 2]
```diff
@@ -308,26 +308,41 @@ std::vector<CQuorumCPtr> CQuorumManager::ScanQuorums(Consensus::LLMQType llmqTyp
...
 {
     std::vector<CQuorumCPtr> result;
+    result.reserve(maxCount);
+
+    auto& params = Params().GetConsensus().llmqs.at(llmqType);
+
+    auto firstQuorumHash = quorumBlockProcessor->GetFirstMinedQuorumHash(llmqType);
+    if (firstQuorumHash.IsNull()) {
+        // no quorum mined yet, avoid scanning the whole chain down to genesis
+        return {};
```
Comment (UdjinM6) [inline src/llmq/quorums.cpp]:
```suggestion
        return result;
```
for consistency?

**Comment by codablock** [comment]
@UdjinM6 applied the suggestions

File: src/llmq/quorums_blockprocessor.cpp [hunk 3]
```diff
@@ -188,6 +189,9 @@ bool CQuorumBlockProcessor::ProcessCommitment(const CBlockIndex* pindex, const C
 
     // Store commitment in DB
     evoDb.Write(std::make_pair(DB_MINED_COMMITMENT, std::make_pair((uint8_t)params.type, quorumHash)), qc);
+    if (!evoDb.Exists(std::make_pair(DB_FIRST_MINED_COMMITMENT, (uint8_t)params.type))) {
+        evoDb.Write(std::make_pair(DB_FIRST_MINED_COMMITMENT, (uint8_t)params.type), quorumHash);
```
[the remark this answers is not in the document]
Comment (codablock) [reply src/llmq/quorums_blockprocessor.cpp]: :see_no_evil: Fixed

**Pull request merged**